testing/tests/DevExpress.ui.widgets.scheduler/common.tests.js

```diff
@@ -330,77 +319,131 @@ QUnit.testStart(function() {
                 allDay: true,
                 text: "Task 1"
             }, {
-                startDate: new Date(2015, 4, 24, 9, 10),
-                endDate: new Date(2015, 4, 24, 11, 1),
+                startDate: new Date(2015, 4, 24, 9),
+                endDate: new Date(2015, 4, 24, 11),
                 allDay: true,
-                recurrenceRule: "FREQ=DAILY;COUNT=2",
+                recurrenceRule: "FREQ=DAILY;COUNT=3",
                 text: "Task 2"
             }],
             maxAppointmentsPerCell: 1,
             currentDate: new Date(2015, 4, 24),
-            displayedAppointmentDataField: "Field",
             views: ["month"],
-            dropDownAppointmentTemplate: function(data) {
-                assert.deepEqual(data.Field.startDate, new Date(2015, 4, 25, 9, 10), "Start date of part is ok");
-                assert.deepEqual(data.Field.endDate, new Date(2015, 4, 25, 11, 1), "End date of part is ok");
+            dropDownAppointmentTemplate: function(itemData, itemIndex, itemElement, targetedAppointmentData) {
+                delete targetedAppointmentData.settings;
+                assert.deepEqual(targetedAppointmentData, {
+                    allDay: true,
+                    endDate: new Date(2015, 4, 25, 9),
+                    recurrenceRule: "FREQ=DAILY;COUNT=3",
+                    startDate: new Date(2015, 4, 25, 9),
+                    text: "Task 2"
+                });
+            },
+            currentView: "month"
+        });
+
+        $(".dx-scheduler-dropdown-appointments").eq(1).dxDropDownMenu("instance").open();
+    });
+
+    QUnit.test("The appointment data argument of the appointment tooltip template is should be an instance", function(assert) {
+        var data = [{
+            startDate: new Date(2015, 4, 24, 9),
+            endDate: new Date(2015, 4, 24, 11),
+            text: "Task"
+        }];
+
+        this.instance.option({
+            dataSource: data,
+            height: 600,
+            currentDate: new Date(2015, 4, 24),
+            currentView: "month",
+            views: ["month"],
+            appointmentTooltipTemplate: function(appointmentData) {
+                assert.equal(data.indexOf(appointmentData), 0);
+            }
+        });
+
+        $(this.instance.$element().find(".dx-scheduler-appointment").eq(0)).trigger("dxclick");
+        this.clock.tick(300);
+    });
+
+    QUnit.test("The appointment data argument of the dropdown appointment template is should be an instance", function(assert) {
+        var data = [{
+            startDate: new Date(2015, 4, 24, 9),
+            endDate: new Date(2015, 4, 24, 11),
+            recurrenceRule: "FREQ=DAILY;COUNT=3",
+            allDay: true,
+            text: "Task 1"
+        }, {
+            startDate: new Date(2015, 4, 24, 9),
+            endDate: new Date(2015, 4, 24, 11),
+            allDay: true,
+            recurrenceRule: "FREQ=DAILY;COUNT=3",
```

Comment:
Is it necessary to create a recurrence entry here? Or, should be recurrence tested against another current date?

Reply:
I agree with you and will remove the recurrenceRule from test data.